test/k8sT/Services.go

```diff
@@ -0,0 +1,69 @@
+package k8sT
+
+import (
+	"fmt"
+
+	"github.com/asaskevich/govalidator"
+	"github.com/cilium/cilium/test/helpers"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	log "github.com/sirupsen/logrus"
+)
+
+var _ = Describe("K8sServicesTest", func() {
+
+	var kubectl *helpers.Kubectl
+	var logger *log.Entry
+	var initilized bool
+	var serviceName string = "app1-service"
+	initilize := func() {
+		if initilized == true {
+			return
+		}
+		logger = log.WithFields(log.Fields{"test": "K8sServiceTest"})
+		logger.Info("Starting")
+
+		kubectl = helpers.CreateKubectl("k8s1", logger)
+		path := fmt.Sprintf("%s/cilium_ds.yaml", kubectl.ManifestsPath())
+		kubectl.Apply(path)
+		_, err := kubectl.WaitforPods("kube-system", "-l k8s-app=cilium", 600)
+		Expect(err).Should(BeNil())
+		initilized = true
+	}
+
+	BeforeEach(func() {
+		initilize()
+	})
+
+	It("Check Service", func() {
+		demoDSPath := fmt.Sprintf("%s/demo.yaml", kubectl.ManifestsPath())
+		kubectl.Apply(demoDSPath)
+		pods, err := kubectl.WaitforPods("default", "-l zgroup=testapp", 300)
+		Expect(pods).Should(BeTrue())
+		Expect(err).Should(BeNil())
+
+		svcIP, err := kubectl.Get(
+			"default", fmt.Sprintf("service %s", serviceName)).Filter("{.spec.clusterIP}")
+		Expect(err).Should(BeNil())
+		Expect(govalidator.IsIP(svcIP.String())).Should(BeTrue())
+
+		status := kubectl.Node.Exec(fmt.Sprintf("curl http://%s/", svcIP))
+		Expect(status.Correct()).Should(BeTrue())
+
+		ciliumPod, err := kubectl.GetCiliumPodOnNode("kube-system", "k8s1")
+		Expect(err).Should(BeNil())
+
+		service := kubectl.CiliumExec(ciliumPod, "cilium service list")
+		Expect(service.Output()).Should(ContainSubstring(svcIP.String()))
+		Expect(service.Correct()).Should(BeTrue())
+
+		kubectl.Delete(demoDSPath)
+	}, 300)
+
+	//FIXME: Check service with IPV6
+	//FIXME: Check the service with cross-node
```

Comment:
I think this is a blocker for merging and that it is critical that we have cross-service tests as is done in `tests/k8s/tests/02-cnp-specs.sh`. cc: @aanm  do you agree?